packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,302 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as s3 from '@aws-cdk/aws-s3';
+import { Construct, IResource, Lazy, Resource, Stack, Token } from '@aws-cdk/core';
+import { CfnDistribution } from './cloudfront.generated';
+import { Origin } from './origin';
+import { CacheBehavior } from './private/cache-behavior';
+
 /**
  * Interface for CloudFront distributions
  */
-export interface IDistribution {
+export interface IDistribution extends IResource {
+  /**
+   * The domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The distribution ID for this distribution.
+   *
+   * @attribute
+   */
+  readonly distributionId: string;
+}
+
+/**
+ * Attributes used to import a Distribution.
+ *
+ * @experimental
+ */
+export interface DistributionAttributes {
   /**
-   * The domain name of the distribution
+   * The generated domain name of the Distribution, such as d111111abcdef8.cloudfront.net.
+   *
+   * @attribute
    */
   readonly domainName: string;
 
   /**
    * The distribution ID for this distribution.
+   *
+   * @attribute
    */
   readonly distributionId: string;
+}
+
+/**
+ * Properties for a Distribution
+ *
+ * @experimental
+ */
+export interface DistributionProps {
+  /**
+   * The default behavior for the distribution.
+   */
+  readonly defaultBehavior: BehaviorOptions;
+
+  /**
+   * Additional behaviors for the distribution, mapped by the pathPattern that specifies which requests to apply the behavior to.
+   *
+   * @default no additional behaviors are added.
+   */
+  readonly additionalBehaviors?: Record<string, BehaviorOptions>;
+
+  /**
+   * A certificate to associate with the distribution. The certificate must be located in N. Virginia (us-east-1).
+   *
+   * @default the CloudFront wildcard certificate (*.cloudfront.net) will be used.
+   */
+  readonly certificate?: acm.ICertificate;
+
+  /**
+   * The price class that corresponds with the maximum price that you want to pay for CloudFront service.
+   * If you specify PriceClass_All, CloudFront responds to requests for your objects from all CloudFront edge locations.
+   * If you specify a price class other than PriceClass_All, CloudFront serves your objects from the CloudFront edge location
+   * that has the lowest latency among the edge locations in your price class.
+   *
+   * @default PRICE_CLASS_ALL
+   */
+  readonly priceClass?: PriceClass;
+
+  /**
+   * How CloudFront should handle requests that are not successful (e.g., PageNotFound).
+   *
+   *
+   * @default - No custom error configuration.
+   */
+  readonly errorConfigurations?: CfnDistribution.CustomErrorResponseProperty[];
```

Comment:
I opted to directly use the Cfn* interface here, as it really is reasonably complete and simple, and opted to do some validation at construction time in terms of combinations of optional properties. An alternative -- that I coded and discarded -- was to create a class for this with the validations baked in, but at the end it seemed less user-friendly for more effort.

Reply:
Not a big fan. We should have full control of this layer, and exposing this would mean subjecting ourselves to CFN modeling. 

You don't have to create a class, simply duplicating the CFN interface and leaving the validations as is is ok too.

Reply:
Yes we do not expose L1 types in L2 as a general rule

Reply:
Got it. One positive side-effect of this is that awslint complained the TTL field needed to be a Duration, which is an improvement over the "plain" number of the underlying Cfn struct.